diff --git a/wpa_supplicant/ctrl_iface.c b/wpa_supplicant/ctrl_iface.c
index 0fac389..349793c 100644
--- a/wpa_supplicant/ctrl_iface.c
+++ b/wpa_supplicant/ctrl_iface.c
@@ -21,6 +21,7 @@
 #include "common/wpa_ctrl.h"
 #ifdef CONFIG_DPP
 #include "common/dpp.h"
+#include "dpp_supplicant.h"
 #endif /* CONFIG_DPP */
 #include "common/ptksa_cache.h"
 #include "crypto/tls.h"
@@ -55,7 +56,6 @@
 #include "offchannel.h"
 #include "drivers/driver.h"
 #include "mesh.h"
-#include "dpp_supplicant.h"
 #include "sme.h"
 
 #ifdef __NetBSD__
diff --git a/wpa_supplicant/events.c b/wpa_supplicant/events.c
index b3f1d37..29b4632 100644
--- a/wpa_supplicant/events.c
+++ b/wpa_supplicant/events.c
@@ -49,7 +49,9 @@
 #include "mesh.h"
 #include "mesh_mpm.h"
 #include "wmm_ac.h"
+#ifdef CONFIG_DPP
 #include "dpp_supplicant.h"
+#endif /* CONFIG_DPP */
 #include "rsn_supp/wpa_i.h"
 
 
diff --git a/wpa_supplicant/wpa_supplicant.c b/wpa_supplicant/wpa_supplicant.c
index 9ef7bb0..8980132 100644
--- a/wpa_supplicant/wpa_supplicant.c
+++ b/wpa_supplicant/wpa_supplicant.c
@@ -63,7 +63,9 @@
 #include "wnm_sta.h"
 #include "wpas_kay.h"
 #include "mesh.h"
+#ifdef CONFIG_DPP
 #include "dpp_supplicant.h"
+#endif /* CONFIG_DPP */
 #ifdef CONFIG_MESH
 #include "ap/ap_config.h"
 #include "ap/hostapd.h"
