diff --git a/hostapd/ctrl_iface.c b/hostapd/ctrl_iface.c
index a2903f2dd0..08bc466fe6 100644
--- a/hostapd/ctrl_iface.c
+++ b/hostapd/ctrl_iface.c
@@ -3469,7 +3469,7 @@ static int hostapd_ctrl_iface_get_capability(struct hostapd_data *hapd,
 }
 
 
-#ifdef ANDROID
+#if defined (ANDROID) || defined (LINUX_EMBEDDED)
 static int hostapd_ctrl_iface_driver_cmd(struct hostapd_data *hapd, char *cmd,
 					 char *buf, size_t buflen)
 {
diff --git a/src/ap/ap_drv_ops.h b/src/ap/ap_drv_ops.h
index 0bfc154647..23ce5433a9 100644
--- a/src/ap/ap_drv_ops.h
+++ b/src/ap/ap_drv_ops.h
@@ -432,7 +432,7 @@ hostapd_drv_set_band(struct hostapd_data *hapd, u32 band_mask)
 	return hapd->driver->set_band(hapd->drv_priv, band_mask);
 }
 
-#ifdef ANDROID
+#if defined(ANDROID) || defined(LINUX_EMBEDDED)
 static inline int hostapd_drv_driver_cmd(struct hostapd_data *hapd,
 					 char *cmd, char *buf, size_t buf_len)
 {
