diff --git a/src/ap/hw_features.c b/src/ap/hw_features.c
index c4619d7..96adc1b 100644
--- a/src/ap/hw_features.c
+++ b/src/ap/hw_features.c
@@ -736,10 +736,11 @@ static int hostapd_is_usable_chans(struct hostapd_iface *iface)
	if (!iface->conf->secondary_channel)
		return 1;

+	if (hostapd_is_usable_chan(iface, iface->conf->channel +
+				   iface->conf->secondary_channel * 4, 0))
+		return 1;
	if (!iface->conf->ht40_plus_minus_allowed)
-		return hostapd_is_usable_chan(
-			iface, iface->conf->channel +
-			iface->conf->secondary_channel * 4, 0);
+		return 0;

	/* Both HT40+ and HT40- are set, pick a valid secondary channel */
	secondary_chan = iface->conf->channel + 4;
