diff --git a/src/ap/hw_features.c b/src/ap/hw_features.c
index 4b66b02f4..e15d9af74 100644
--- a/src/ap/hw_features.c
+++ b/src/ap/hw_features.c
@@ -927,6 +927,11 @@ static int hostapd_is_usable_chans(struct hostapd_iface *iface)
 		    (pri_chan->allowed_bw & HOSTAPD_CHAN_WIDTH_40M))
 			return 1;
 	}
+
+	if (hostapd_is_usable_chan(iface, iface->conf->channel +
+				   iface->conf->secondary_channel * 4, 0))
+		return 1;
+
 	if (!iface->conf->ht40_plus_minus_allowed)
 		return 0;
 
